lib/BackendClient.js

```diff
@@ -433,6 +439,35 @@ var BackendClient = (function(){
         return deferred.promise;
     }
 
+    function sendBeacon(entryId, partnerId, data, beaconTag) {
+        let tag = beaconPrefix + '_' + beaconTag;
+        let deferred = Q.defer();
+        logger.debug(`[${entryId}] Calling beacon.add, Tag: [${tag}]`);
+
+        let KalturaBeacon = new kalturaVO.KalturaBeacon();
+        KalturaBeacon.relatedObjectType = kalturaTypes.BeaconObjectTypes.ENTRY_SERVER_NODE_BEACON;
+        KalturaBeacon.eventType = tag;
+        KalturaBeacon.objectId = entryId;
+        KalturaBeacon.privateData = JSON.stringify(data);
+
+        //Impersonate call to the entry's partnerId to execute the call in the partner's context
+        client.setPartnerId(partnerId);
```

Comment:
this is bad... since from this point onward ALL API requests will be done with this specific partnerId...